### PR TITLE
docs - add information about view dependency

### PR DESCRIPTION
**Pull request by mkiyama**
-example queries to find dependency information
-catalog table information
-best practices

This will be backported to 6X_STABLE and 5X_STABLE

Link to HTML output on a temporary GPDB draft doc website

Creating and Managing Views
https://docs-msk-gpdb6-dev.cfapps.io/7-0/admin_guide/ddl/ddl-view.html

Working with View Dependencies - Example Queries
https://docs-msk-gpdb6-dev.cfapps.io/7-0/admin_guide/ddl/ddl-view-find-depend.html

Best Practices
https://docs-msk-gpdb6-dev.cfapps.io/7-0/admin_guide/ddl/ddl-view-best-practices.html

About View Storage in Greenplum Database
https://docs-msk-gpdb6-dev.cfapps.io/7-0/admin_guide/ddl/ddl-view-storage.html

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
+<topic id="topic_id4_pfb_3lb">
+  <title>About View Storage in Greenplum Database</title>
+  <body>
+    <p>A view is similar a table, both are relations, that is "something with columns". All such
```
Comment (litzell) [inline gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml:6]: similar a -> similar to a

Maybe the comma after table could be a colon or mdash.

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    <ul id="ul_mrc_15l_flb">
+      <li>Object names are resolved during <codeph>CREATE VIEW</codeph>, so the current setting of
+          <codeph>search_path</codeph> affects the view definition.</li>
+      <li>Objects are referred to by their internal immutable object ID rather than by their name.
+        Consequently, renaming an object or column referenced in a view definition can be performed
+        without dropping the view.</li>
+      <li>Greenplum Database can determine exactly which objects are used in the view definition, so
+        it can add dependencies on them.</li>
+    </ul>
+    <p>Note that the way Greenplum Database handles views quite different from the way Greenplum
```
Comment (litzell) [inline gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml:34]: views quite -> views is quite

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      Database handles functions: function bodies are stored as strings and are not parsed when they
+      are created. Consequently, Greenplum Database does know on which objects a given function
```
Comment (litzell) [inline gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml:36]: does know -> does not know

File: gpdb-doc/dita/admin_guide/ddl/ddl-view.dita [hunk 4]
```diff
@@ -29,10 +36,10 @@
         <codeblock>DROP VIEW topten;
 </codeblock>
       </p>
-      <p>The <codeph>DROP VIEW ... CASCADE</codeph> command also removes all dependent objects. As
-      example, if another view depends on the view which is about to be dropped, the other view
-      will be dropped as well. Without the <codeph>CASCADE</codeph> option, the
-      <codeph>DROP VIEW</codeph> command will fail.</p>
+      <p>The <codeph>DROP VIEW...CASCADE</codeph> command also removes all dependent objects. As
+        example, if another view depends on the view which is about to be dropped, the other view
```
Comment (litzell) [inline gpdb-doc/dita/admin_guide/ddl/ddl-view.dita:40]: As example -> As an example

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
+<topic id="topic_dn2_52b_3lb">
+  <title>Best Practices when Creating Views</title>
+  <body>
+    <p>When defining and using view, remember that a view is just an SQL statement and is replaced
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml:6]: using view -> using a view

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+ v1         | t1         | id
+ v2         | t1         | id
+ mytest.vt1 | t1         | id
+ mytest.v2a | t1         | id
+(4 rows)</codeblock>
+      </section>
+      <section id="view_schema">
+        <title>Listing View Schemas</title>
+        <p>If you have created views in multiple schemas, you can also list views and the view's
+          schema the table referenced by the view. The query retrieves schema from the catalog table
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml:147]: There's something missing in this sentence around "view's schema the table".  Maybe it should read "the view's schema using the table"?  Not sure.

Comment (dyozie) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml:147]: retries schema -> retrieves the schema

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+      <p>For detailed information about the system catalog tables that store view information, see
+          <xref href="ddl-view-storage.xml"/>.</p>
+      <section id="depend_table">
+        <title>Finding Direct View Dependencies on a Table</title>
+        <p>To find out which views directly depend on table <codeph>t1</codeph>, create a query that
+          performs a join among the catalog tables that contain the dependency information, and
+          qualify the query to return only view dependencies.</p>
+        <codeblock>SELECT v.oid::regclass AS view,
+  d.refobjid::regclass AS ref_object    -- name of table
+  -- d.refobjid::regproc as ref_object  -- name of function
```
Comment (jimmyyih) [inline gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml:59]: Capitalize `as` keyword for consistency.

**Comment by mkiyama** [comment]
@ivannovick @lmugnano 
If you have time, please review.

Thanks for your help

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      clause of the <codeph>SELECT</codeph> command instead of creating a seldom-used view.</p>
+    <p>In general, these uses do not require nesting views, that is, defining views based on other
+      views.</p>
+    <p>These are two patterns of creating views that tend to be problematic because the view's SQL
+      is used during query execution.</p>
+    <ul id="ul_krc_15l_flb">
+      <li>
+        <p>Defining many layers of views so that your final queries look deceptively simple.</p>
+        <p>Problems arise when you try to enhance or troubleshoot queries that use the views, for
+          example by examining at the execution plan. The query's execution plan tends to be
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+          complicated and is difficult to understand and how to improve it.</p>
+      </li>
+      <li>
+        <p>Defining a denormalized "world" view. A view that is join of a large number of database
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+    <body>
+      <p>The following example queries list view information on dependencies on tables and columns. </p>
+      <ul id="ul_xg2_bls_jlb">
+        <li><xref href="#topic_find_v_examples/depend_table" format="dita"/></li>
+        <li><xref href="#topic_find_v_examples/direct_column" format="dita"/></li>
+        <li><xref href="#topic_find_v_examples/view_schema" format="dita"/></li>
+        <li><xref href="#topic_find_v_examples/view_defs" format="dita"/></li>
+        <li><xref href="#topic_find_v_examples/nested_views" format="dita"/></li>
+      </ul>
+      <p>The exanple output is based on the <xref href="#topic_find_v_examples/example_data"
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+ mytest.vt1 | t1
+ mytest.v2a | t1
+ mytest.v2a | t1
+(7 rows)</codeblock>
+        <p>The query performs casts to the <codeph>regclass</codeph> object identifier type. For
+          information about object identifier types, see the PostgeSQL documentation on <xref
+            href="https://www.postgresql.org/docs/9.4/datatype-oid.html" format="html">Object
+            Identifier Types</xref>.</p>
+        <p>In some cases, the views are listed multiple times because the view references multiple
+          table columns. You can remove those duplicates using <codeph>DISTINCT</codeph>..</p>
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+ v3   | t1         |       2 | CREATE VIEW v3 AS  SELECT (t1.val || f()) +
+      |            |         |    FROM t1;
+ v2   | t1         |       2 | CREATE VIEW v2 AS  SELECT t1.val          +
+      |            |         |    FROM (t1                               +
+      |            |         |      JOIN v1 USING (id));
+(7 rows)</codeblock>
+      </section>
+      <section id="nested_views">
+        <title>Listing Nested Views</title>
+        <p>This CTE query lists information about views that reference on another view.</p>
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml [hunk 13]
```diff
@@ -0,0 +1,323 @@
...
+ mytest.v2a | t1         | id
+(4 rows)</codeblock>
+      </section>
+      <section id="view_schema">
+        <title>Listing View Schemas</title>
+        <p>If you have created views in multiple schemas, you can also list views and the view's
+          schema the table referenced by the view. The query retrieves schema from the catalog table
+            <codeph>pg_namespace</codeph> and excludes the system schemas
+            <codeph>pg_catalog</codeph>, <codeph>information_schema</codeph>, and
+            <codeph>gp_toolkit</codeph>. Also. the query does not list a view if the view refers to
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml (hunk 2 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml (hunk 3 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view.dita (hunk 4 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view.dita:40]: fixed

Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view.dita:40]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml (hunk 5 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-best-practices.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml (hunk 6 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+      depends.</p>
+    <section>
+      <title>Where View Dependency Information is Stored</title>
+      <p>These system catalog tables contain the information that are used to determine the tables
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml (hunk 7 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml [hunk 15]
```diff
@@ -0,0 +1,323 @@
...
+        <p>In some cases, the views are listed multiple times because the view references multiple
+          table columns. You can remove those duplicates using <codeph>DISTINCT</codeph>..</p>
+        <p>You can alter the query to find views with direct dependencies on the function
+            <codeph>f</codeph>. <ul id="ul_m2y_mn2_jlb">
+            <li>In the <codeph>SELECT</codeph> clause replace the name of the table
+                <codeph>d.refobjid::regclass as ref_object</codeph> with the name of the function
+                <codeph>d.refobjid::regproc as ref_object</codeph></li>
+            <li>In the <codeph>WHERE</codeph> clause replace the catalog of the referenced object
+              from <codeph>d.refclassid = 'pg_class'::regclass</codeph> for tables, to
+                <codeph>d.refclassid = 'pg_proc'::regclass</codeph> for a procedures (functions).
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml (hunk 1 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-storage.xml]: fixed

File: gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml (hunk 7 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/ddl/ddl-view-find-depend.xml]: @jimmyyih  Thanks for reviewing.

**Pull request merged**